### ISSUE TITLE
Thumbnail move to external card

### ISSUE THREAD

**Issue by FlipCook** [eleybourn/Book-Catalogue#796]
I've a 174Mb worth of thumbnails sitting on my internal memory. Please for my sanity is there a way to move them to an external card?

**Comment by JKS258** [comment]
I have the same issue with the books I've been buying since I first purchased your most excellent program not long after it came out. It works perfectly and does everything I need, with one potentially fatal flaw I think I'm getting ready to experience. Just the books I have on my internal drive take up a huge amount of space, while I have a 250GB SD card sitting there not close to full. However, Im finally getting ready to add my entire library of roughly 2000 books to your app, and if what I have in the database now is using up such a large amount of internal memory, adding 2000 more books to it will likely cause my phone to burst into flames! :-) Like the use above, there's GOT to be some way to store at least the thumbnails, if not the entire program on the SD card. If not, though your program is without a doubt the best book database out there, I may be forced to see if I can find a second-rate substitute before I start this daunting project. I REALLY don't want to do that, since again I love what you've done, but I simply won't have enough internal memory to store the incoming thumbnails. 

I'm hoping to start this project in a couple of weeks or less, so some timely help would be greatly appreciated. 

Thanks again for the great App, and I'm sure hoping your have a solution. 

All the best, Jay...

**Comment by Michael-Logies** [comment]
I would buy a better smartphone with more internal space, e. g. an Honor 7X, 46 GB internal memory are free. How much space how many books are using at the moment on your smartphone?

**Comment by FlipCook** [comment]
So rather than change a string  in the program that allows access to an external card you're suggesting buying a brand new phone?

**Comment by Michael-Logies** [comment]
For me buying an Honor 7X (I have an Honor 6X) as a smartphone with more internal memory would be much cheaper than to hire a programmer for changing "Book Catalogue" (and no, it`s not as easy as "change a string"). I assume that the file sizes you have on your smartphone within BC now are quite modest, so I assume it`s an old (older than 2 years) smartphone and the easiest solution may be to replace it. My Honor 6X works very well for "Book Catalogue".

**Comment by JKS258** [comment]
I agree with FlipCook. While I've overly-stated my love for the Book Catalogue App, to design a database that stores images of any sort ONLY on Internal memory is a bit short-sighted to start with, since free Internal memory is the key to a smooth and speedy phone. I also agree that for a programmer to suggest  to a customer that like myself might have over 200 Apps on their phone, and a highly customized user interface, that the best solution to THEIR programming oversight would be just to get a new phone at a cost that eclipses the cost of the App in question, isn't the ideal answer to a serious and universal customer problem. I've only got 266 books in my database now, but getting ready to add my entire library of 2000 or more books at least, where if I do, my phone will be locked up from lack of ANY Internal memory. 

Since I need to start this project in the next couple of weeks, is it safe to assume that Book Catalogue is NOT going to make storing thumbnails on the SD card an option anytime in the near future? I've found many other book database Apps, one that's more that acceptable, though is still no comparison to Book Catalogue, and it WILL store the thumbnails on the SD card along with most of the other options offered by Book Catalogue, at least the ones I use the most, so would it be your advice to go ahead and switch from my PAID Book Catalogue App to a competing program? What a shame to lose a loyal customer over a VERY legitimate and universal problem, instead of adapting the program to where it should have been to start with. 

Respectfully submitted, Jay... 
Registered User

**Comment by Michael-Logies** [comment]
I`m not a programmer, I`m only a user of BC. BC is free software you don`t have to pay for it.

**Comment by JKS258** [comment]
Sorry about that. I thought you were an administrator. I've had it so long,
I "thought" I paid for it, maybe before it was open source? Not sure, but
do love it and do wish the thumbnails could be stored on the SD card. Looks
like I'm going to have to abandon it though and switch to a lesser program
simply for that one option. For the actual creator of such an otherwise
great program, that seems like such a shame. Thanks for your input though.
Best, Jay...


J.K. (Jay) Statham
1852 Banking Street, Suite 9201
Greensboro, NC 27408
Cell: 336-404-0725
Email: JKS258@GMail.com

**Comment by eleybourn** [comment]
Hi all, 

I'm one of the original authors of the application. We wrote this entirely as volunteers to try and solve a need (and thank you for the positive feedback). And, yes, it's always been free & open source - we've never charged for it. 

In terms of storing on the sdcard, most people just use the "Move to SDCard" option in their phone settings. However, I see that the Honor doesn't have that option. I don't have an Honor, so I've had a look online at various sources. I found this which suggests you can set apps to install on an sdcard. https://www.youtube.com/watch?v=5jHB2g6Lz8M. However you would need to take a backup, uninstall and reinstall BookCatalogue.

The other option is to occasionally go through and bulk-reduce the filesize of the thumbnails. I found a few quick tutorials about that online. e.g. https://www.youtube.com/watch?v=19GbI6oFrW0 -> you'd just need to connect the phone to your computer. 

Does that help?

**Comment by eleybourn** [comment]
Or something like this: https://www.raymond.cc/blog/most-convenient-way-to-automatically-resize-photos-for-email-im-blogging-and-web-galleries/

**Comment by Michael-Logies** [comment]
Evan, thanks for commenting! Could you write about the further plans for BC? Is it  in maintenance mode or are new features under development?
I`m a new user of BC and the only feature I`m missing is "Bulk move of books to Bookshelf", which some people would have liked 8 years ago (https://github.com/eleybourn/Book-Catalogue/issues/90). I know that I should be able to circumvent this limitation by exporting, manipulating in a spreadsheet and reimporting, but I would like to avoid that - but this may mean a bigger redesign for BC, how probable is this? (The newer Honors (e. g. 7X or my 6X) have big enough internal memory and should not need moving the library to the SD card, that`s why I recommended them, they are good & cheap.) Thanks! Michael

**Comment by JKS258** [comment]
Thanks for weighing in on this issue, and thanks again for such a great
product. The thumbnail issue is likely only a problem for those of us with
actual libraries, or a huge number of books we'd like to catalog, in my
case roughly 2500 books. I'm not sure if your suggestions will help, but
I'll check them out and let your know. I know for a fact that "Move to
SDCard" option works in a lot of cases, but only until an App is updated.
Once it updates, which on Android seems likely every day (actually is every
2 weeks or so for most), it moves the App right back to the Internal drive.
I use an App called AppMgr Pro III by AZSoft, which they claim with the
paid version, which I have, once the Apps are moved, they will NOT be moved
back when the Apps are updated, but only on rooted phones, which I'm about
to do to see if that does the trick.

Some Apps either automatically, or give you the option to install on the SD
Card, though Book Catalogue does not. However, I'll check out the video you
sent to see if there's a way to trick it to install on the SD Card, which
would be most excellent!

I have already bulk reduced the thumbnail sizes, though I don't know how
small they can be and still be clear and readable. Maybe that's shown in
your other link, but if not and you know the optimum size, that might help.

I'll report back if the rooting option worked to prevent Apps from being
moved back, or if your video link below does the trick. As I've said many
times, your App is WAY too good to abandon if I don't have to, and I'm
willing to try most anything to keep from having to go to another App
(except buy a new phone), but when I add another 2000+ books to the
database and if it can't be moved to the SD card, no matter how small the
images, it will still use up too much valuable internal memory.

Great thanks again for the excellent program. Short that one detail of the
SD Card storage, it's by a million miles that best book catalog App out
there. I have a friend that's been a librarian for 35 years and she said
the same thing about your App, and was quite impressed with all you could
do with it right in your hand, where you'd have instant access to your
entire library on demand. That's particularly great again when you have so
MANY books, because there have been times that I've bought the same book I
already own because I forgot I had it and it wasn't in App yet, which is
one of the reasons I'm about to add 100% of my books to the App. I'm moving
and figured this was the best time to do it if it's ever going to get done.

Thanks again for your help.

All the best, Jay...

**Comment by eleybourn** [comment]
I'll be honest, it's kind of fallen into maintenance mode by default. Both myself and Philip (who are the two volunteer developers) have become too busy with our day-jobs & families and so we haven't had the time to make any significant changes to BC. Most of the app updates have been bug fixes or minor future proofing (e.g. changing calls from http to https).

@Michael-Logies I can't believe some of this is 8 years old now. Now I'm feeling old :-) The easiest way to bulk move is to export to CSV, edit in Excel, and then import from CSV. It's not ideal, but it'll work.

@JKS258 let me know how/if AppMgr works.

**Comment by JKS258** [comment]
Hi Eley,

I still haven't rooted my phone and tried the App Mgr III claim that the
App will not be moved back when updated, as they all do. However, I did
check to see what the Storage setting was for Book Catalogue and it does in
fact say it's been moved to the SD card, but I think that's only the App
itself. The thumbnails still reside in Internal Memory, and aren't
duplicated on the SD Card (indicating the Internal memory images are
orphans and can be deleted. So apparently though the App itself can be
moved, the thumnails cannot, unless you know a workaround.

Now it seems unlikely that rooting will help since your coding is what
determines the location of thumbnail storage. I didn't see an option inside
the App that allows you to redirect the image storage, so I guess we're
back to zero unless you have anymore rabbits in your hat! ☺️

Thanks again, Jay...
